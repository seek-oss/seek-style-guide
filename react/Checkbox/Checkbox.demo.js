import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Checkbox, { STANDARD, BUTTON } from './Checkbox';
import * as sketch from './Checkbox.sketch';
import demoStyles from './Checkbox.demo.less';
import fieldMessageOptions from '../FieldMessage/FieldMessage.demoFragment';

class CheckboxContainer extends Component {
  static propTypes = {
    component: PropTypes.func.isRequired,
    componentProps: PropTypes.object.isRequired
  };

  constructor() {
    super();

    this.state = {
      checked1: false,
      checked2: false
    };
  }

  handleChange1 = event => {
    this.setState({
      checked1: event.target.checked
    });
  };

  handleChange2 = event => {
    this.setState({
      checked2: event.target.checked
    });
  };

  render() {
    const { component: DemoComponent, componentProps } = this.props;
    const { checked1, checked2 } = this.state;

    return (
      <div
        className={classnames({
          [demoStyles.root_isStandard]: componentProps.type === STANDARD,
          [demoStyles.root_isButton]: componentProps.type === BUTTON
        })}
      >
        <DemoComponent
          {...componentProps}
          id="fullTime"
          label="Full time"
          checked={checked1}
          onChange={this.handleChange1}
        />
        <DemoComponent
          {...componentProps}
          id="partTime"
          label="Part time / casual / vacation"
          checked={checked2}
          onChange={this.handleChange2}
        />
      </div>
    );
  }
}

export default {
  route: '/checkbox',
  title: 'Checkbox',
  category: 'Form',
  component: Checkbox,
  container: CheckboxContainer,
  sketch,
  initialProps: {
    type: 'standard',
    // Documentation only:
    label: 'Full Time',
    checked: false,
    onChange: () => {}
  },
  options: [
    {
      label: 'Type',
      type: 'radio',
      states: [
        {
          label: 'Standard',
          transformProps: props => props
        },
        {
          label: 'Button',
          transformProps: props => ({
            ...props,
            type: 'button'
          })
        }
      ]
    },
    ...fieldMessageOptions
  ]
};
