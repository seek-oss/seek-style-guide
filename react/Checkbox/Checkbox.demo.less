@import "~seek-style-guide/theme";

.root_isStandard {
  display: flex;
  flex-direction: column;
  width: 220px;
}

.root_isButton > :not(*:last-child) {
  margin-right: @gutter-width / 2;
}
