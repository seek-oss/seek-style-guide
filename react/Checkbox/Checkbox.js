import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import styles from './Checkbox.less';
import CheckMarkIcon from '../CheckMarkIcon/CheckMarkIcon';
import classnames from 'classnames';
import FieldMessage from '../FieldMessage/FieldMessage';
import Text from '../Text/Text';
import { TONE } from '../private/tone';

export const STANDARD = 'standard';
export const BUTTON = 'button';

function combineClassNames(props = {}, ...classNames) {
  const { className, ...restProps } = props;

  return {
    className: classnames.apply(null, [...classNames, className]), // eslint-disable-line no-useless-call
    ...restProps
  };
}

export default class Checkbox extends Component {
  static displayName = 'Checkbox';

  static propTypes = {
    id: PropTypes.string.isRequired,
    label: PropTypes.node.isRequired,
    value: PropTypes.string,
    className: PropTypes.string,
    checked: PropTypes.bool.isRequired,
    onChange: PropTypes.func.isRequired,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func,
    inputProps: PropTypes.object,
    type: PropTypes.oneOf([STANDARD, BUTTON]),
    valid: PropTypes.bool,
    message: PropTypes.oneOfType([PropTypes.oneOf([false]), PropTypes.node]),
    messageProps: PropTypes.object,
    tone: PropTypes.oneOf([
      TONE.POSITIVE,
      TONE.INFO,
      TONE.CRITICAL,
      TONE.NEUTRAL
    ])
  };

  static defaultProps = {
    className: '',
    checked: false,
    inputProps: {},
    type: STANDARD
  };

  renderButton(label) {
    return (
      <Text
        component="span"
        baseline={false}
        raw={true}
        className={styles.button}
      >
        {label}
      </Text>
    );
  }

  renderStandard(label) {
    return (
      <Fragment>
        <div className={styles.checkbox}>
          <CheckMarkIcon
            className={styles.checkMark}
            svgClassName={classnames(
              styles.checkMarkSvg,
              styles.checkMarkSvg_isHover
            )}
          />
          <CheckMarkIcon
            svgClassName={classnames(
              styles.checkMarkSvg,
              styles.checkMarkSvg_isSelected
            )}
            className={styles.checkMark}
          />
        </div>
        <Text
          component="span"
          baseline={false}
          raw={true}
          className={styles.labelText}
        >
          {label}
        </Text>
      </Fragment>
    );
  }

  renderLabel() {
    const { id, label, type } = this.props;

    return (
      <label htmlFor={id} className={styles.label}>
        {type === STANDARD
          ? this.renderStandard(label)
          : this.renderButton(label)}
      </label>
    );
  }

  renderMessage(id, valid, message, messageProps, tone) {
    return (
      <FieldMessage
        {...{
          id: `${id}-message`,
          message,
          messageProps,
          tone,
          ...(tone ? {} : { valid })
        }}
      />
    );
  }

  renderInput() {
    const {
      id,
      value,
      checked,
      onChange,
      onFocus,
      onBlur,
      inputProps,
      type
    } = this.props;

    const allInputProps = {
      id,
      value,
      checked,
      onChange,
      onFocus,
      onBlur,
      ...combineClassNames(inputProps, {
        [styles.input]: true,
        [styles.input_isCheckbox]: type === STANDARD,
        [styles.input_isButton]: type === BUTTON
      }),
      type: 'checkbox'
    };

    return <input {...allInputProps} />;
  }

  render() {
    const { className, id, valid, message, messageProps, tone } = this.props;

    const rootClassNames = classnames({
      [styles.root]: true,
      [styles.invalid]:
        typeof tone !== 'undefined' ? tone === TONE.CRITICAL : valid === false,
      [className]: className
    });

    return (
      <div className={rootClassNames}>
        {this.renderInput()}
        {this.renderLabel()}
        {this.renderMessage(id, valid, message, messageProps, tone)}
      </div>
    );
  }
}
