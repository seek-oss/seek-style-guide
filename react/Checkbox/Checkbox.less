@import (reference) "~seek-style-guide/theme";

@checkboxSize: @grid-row-height * 3;

.activeFocusReset() {
  .input:not([readonly]):not([disabled]):focus + .label &,
  .input:not([readonly]):not([disabled]):active + .label & {
    outline: none;
    border-color: transparent;
    .focus();
  }
}

.root {
  display: inline-block;
}

.input {
  opacity: 0;
  position: absolute;
}

.label {
  display: block;
  height: 100%;

  :checked + & .labelContainer {
    font-weight: @sk-bold;
  }
}

.standard {
  .touchableText(@standard-type-scale-mobile);
  cursor: pointer;
  display: flex;
  align-items: center;
  @media @desktop {
    .touchableText(@standard-type-scale);
  }

  &.isMultiline {
    .standardTextResponsive(@baseline: false);
    align-items: start;
  }
}

.checkBox {
  .activeFocusReset();

  background-color: @sk-white;
  width: @checkboxSize;
  min-width: @checkboxSize;
  height: @checkboxSize;
  border: @field-border-width solid @sk-mid-gray-light;
  border-radius: @field-border-radius;
  margin-right: @field-gutter-width;
  padding: @field-border-radius;
  fill: @sk-pink;
}

.checkMark {
  display: block;
  width: 100%;
  height: 100%;
  fill: @sk-form-accent;
  opacity: 0;

  .input:checked + .label & {
    opacity: 1;
  }
}

.button {
  .activeFocusReset();
  .touchableText(@standard-type-scale-mobile);
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: @sk-gray-light;
  color: @sk-charcoal;
  border: 1px solid @sk-mid-gray-light;
  border-radius: 2px;
  cursor: pointer;
  @media @desktop {
    .touchableText(@standard-type-scale);
  }

  .input:checked + .label & {
    font-weight: @sk-bold;
    background-color: @sk-white;
    border-bottom: 1px solid @sk-pink;
  }
}
