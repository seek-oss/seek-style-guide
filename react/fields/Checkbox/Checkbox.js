import React, { Component, PropTypes } from 'react';
import styles from './Checkbox.less';
import CheckMarkIcon from '../../icons/CheckMarkIcon/CheckMarkIcon';
import classnames from 'classnames';

function combineClassNames(props = {}, ...classNames) {
  const { className, ...restProps } = props;

  return {
    className: classnames.apply(null, [...classNames, className]), // eslint-disable-line no-useless-call
    ...restProps
  };
}

export default class Checkbox extends Component {
  static displayName = 'Checkbox';

  static propTypes = {
    id: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    className: PropTypes.string,
    inputProps: PropTypes.shape({
      onChange: PropTypes.func.isRequired,
      checked: PropTypes.bool.isRequired
    })
  }

  static defaultProps = {
    className: '',
    inputProps: {
      checked: false
    }
  };

  renderLabel() {
    const { label, id } = this.props;

    return (
      <label className={styles.label} htmlFor={id}>
        <CheckMarkIcon svgClassName={styles.checkMark} className={styles.checkBox} />
        <span>{label}</span>
      </label>
    );
  }

  renderInput() {
    const { inputProps, id } = this.props;

    const allInputProps = {
      ...combineClassNames(inputProps, styles.input),
      type: 'checkbox',
      id
    };

    return (
      <input {...allInputProps} />
    );
  }

  render() {
    const { className } = this.props;

    const rootClassNames = classnames({
      [styles.root]: true,
      [className]: className
    });

    return (
      <div className={rootClassNames}>
        { this.renderInput() }
        { this.renderLabel() }
      </div>
    );
  }
}
