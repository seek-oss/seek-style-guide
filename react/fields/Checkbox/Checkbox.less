@import (reference) "~seek-style-guide/theme";

.root {
  display: inline-block;
}

.input {
  opacity: 0;
  position: absolute;
}

.standard {
  .touchableText(@standard-type-scale);

  cursor: pointer;
  display: flex;
  align-items: center;
}

.checkBox {
  width: @grid-row-height * 3;
  height: @grid-row-height * 3;
  border: @field-border-width solid @sk-mid-gray-light;
  border-radius: @field-border-radius;
  margin-right: @field-gutter-width;
  padding: @field-border-radius;
  fill: @sk-pink;

  .input:not([readonly]):not([disabled]):focus + .label &, .rootFocus & {
    outline: none;
    border-color: transparent;
    .focus();
  }
}

.checkMark {
  display: block;
  width: 100%;
  height: 100%;
  fill: @sk-pink;
  opacity: 0;

  .input:checked + .label & {
    opacity: 1;
  }
}

.button {
  .touchableText(@standard-type-scale);

  display: flex;
  align-items: center;
  justify-content: center;
  background-color: @sk-gray-light;
  color: @sk-charcoal;
  border: 1px solid @sk-mid-gray-light;
  border-radius: 2px;
  cursor: pointer;

  .input:checked + .label & {
    font-weight: @sk-bold;
    background-color: @sk-white;
    border-bottom: 1px solid @sk-pink;
  }

  .input:not([readonly]):not([disabled]):focus + .label &, .rootFocus & {
    outline: none;
    border-color: transparent;
    .focus();
  }
}
