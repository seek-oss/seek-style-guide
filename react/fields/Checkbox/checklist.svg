<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 16.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
 viewBox="90.471 302.809 823 823" enable-background="new 90.471 302.809 823 823"
	 xml:space="preserve">
<g id="icomoon-ignore">
</g>
<g>
	<g>
		<polygon points="365.407,1022.185 90.677,747.457 158.56,679.575 365.406,886.421 845.383,406.432 913.266,474.314 		"/>
	</g>
</g>
</svg>
