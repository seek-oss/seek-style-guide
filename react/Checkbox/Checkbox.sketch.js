import React from 'react';
import PropTypes from 'prop-types';
import Checkbox from './Checkbox';
import styles from './Checkbox.demo.less';

const Container = ({ children }) => (
  <div className={styles.root}>{children}</div>
);
Container.propTypes = {
  children: PropTypes.node.isRequired
};

const noop = () => {};

export const symbols = {
  'Checkbox/Standard/Checked': (
    <Container>
      <Checkbox
        id="checkbox2"
        label="Checkbox"
        checked={true}
        onChange={noop}
      />
    </Container>
  ),
  'Checkbox/Standard/Unchecked': (
    <Container>
      <Checkbox
        id="checkbox1"
        label="Checkbox"
        checked={false}
        onChange={noop}
      />
    </Container>
  ),
  'Checkbox/Button/Checked': (
    <Container>
      <Checkbox
        id="checkbox2"
        label="Checkbox"
        type="button"
        checked={true}
        onChange={noop}
      />
    </Container>
  ),
  'Checkbox/Button/Unchecked': (
    <Container>
      <Checkbox
        id="checkbox2"
        label="Checkbox"
        type="button"
        checked={false}
        onChange={noop}
      />
    </Container>
  )
};
