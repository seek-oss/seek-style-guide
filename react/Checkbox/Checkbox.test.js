import chai, { expect } from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
import React from 'react';
import {
  Simulate,
  renderIntoDocument,
  scryRenderedDOMComponentsWithClass,
  findRenderedDOMComponentWithClass
} from 'react-dom/test-utils';
import { createRenderer } from 'react-test-renderer/shallow';
import SyntheticEvent from 'react-dom/lib/SyntheticEvent';
import { findAllWithClass } from 'react-shallow-testutils';
import Checkbox from './Checkbox';

chai.use(sinonChai);

const eventMatcher = sinon.match.instanceOf(SyntheticEvent);

const renderer = createRenderer();

describe('Checkbox', () => {
  let element, checkbox, rootElement, input, label, errors, standard, button;

  beforeEach(() => {
    errors = [];

    sinon.stub(console, 'error', errorMessage => {
      errors.push(errorMessage);
    });
  });

  afterEach(() => {
    console.error.restore();
  });

  function render(jsx) {
    element = jsx;
    checkbox = renderer.render(element);
    input = findAllWithClass(checkbox, 'input')[0] || null;
    label = findAllWithClass(checkbox, 'label')[0] || null;
    standard = findAllWithClass(checkbox, 'standard')[0] || null;
    button = findAllWithClass(checkbox, 'button')[0] || null;
  }

  function renderToDom(jsx) {
    element = jsx;
    checkbox = renderIntoDocument(element);
    rootElement = scryRenderedDOMComponentsWithClass(checkbox, 'root')[0];
    label = scryRenderedDOMComponentsWithClass(checkbox, 'label')[0];
    input = findRenderedDOMComponentWithClass(checkbox, 'input');
  }

  it('should have a displayName', () => {
    render(<Checkbox />);
    expect(element.type.displayName).to.equal('Checkbox');
  });

  it('should pass through id', () => {
    const id = 'test';
    renderToDom(<Checkbox id={id} />);
    expect(input.id).to.equal(id);
  });

  it('should pass through className', () => {
    const className = 'test';
    renderToDom(<Checkbox className={className} />);
    expect(rootElement.classList.contains('test')).to.equal(true);
  });

  it('should have `htmlFor` equal to `id` when `label` is specified', () => {
    render(<Checkbox id="still-in-role" label="Still in role" />);
    expect(label.props).to.contain.keys({ htmlFor: 'still-in-role' });
  });

  describe('when type is not set', () => {
    it('should render standard checkbox', () => {
      render(<Checkbox />);
      expect(standard).to.not.equal(null);
    });
  });

  describe('when type is standard', () => {
    it('should render standard checkbox', () => {
      render(<Checkbox type="standard" />);
      expect(standard).to.not.equal(null);
    });
  });

  describe('when type is button', () => {
    it('should render button checkbox', () => {
      render(<Checkbox type="button" />);
      expect(button).to.not.equal(null);
    });
  });

  describe('inputProps', () => {
    it('should invoke the onChange handler when touched', () => {
      const handleChange = sinon.spy();
      handleChange.reset();
      const props = {
        id: 'still-in-role',
        label: 'still in role',
        inputProps: {
          onChange: handleChange,
          checked: false
        }
      };
      renderToDom(<Checkbox {...props} />);

      Simulate.change(input, { 'target': { 'checked': true } });
      expect(handleChange.calledOnce).to.equal(true);
      expect(handleChange).to.be.calledWith(eventMatcher);
    });

    it('should pass through className', () => {
      renderToDom(<Checkbox id='test' inputProps={{ checked: true }} />);
      expect(input.checked).to.equal(true);
    });

    it('should pass through other props to the input', () => {
      render(<Checkbox inputProps={{ 'data-automation': 'first-name-field' }} />);
      expect(input.props['data-automation']).to.equal('first-name-field');
    });
  });
});
