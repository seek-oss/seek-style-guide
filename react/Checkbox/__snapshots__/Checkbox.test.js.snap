// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox FieldMessage should render field message when passed 1`] = `
<div
  className="Checkbox__root Checkbox__invalid"
>
  <input
    checked={false}
    className="Checkbox__input Checkbox__input_isCheckbox"
    id="testCheckbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="Checkbox__label"
    htmlFor="testCheckbox"
  >
    <div
      className="Checkbox__checkbox"
    >
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isHover"
      />
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isSelected"
      />
    </div>
    <Text
      baseline={false}
      className="Checkbox__labelText"
      component="span"
      raw={true}
    >
      Still in role
    </Text>
  </label>
  <FieldMessage
    id="testCheckbox-message"
    message="Field is invalid"
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
    valid={false}
  />
</div>
`;

exports[`Checkbox inputProps should pass through other props to the input 1`] = `
<div
  className="Checkbox__root"
>
  <input
    checked={false}
    className="Checkbox__input Checkbox__input_isCheckbox"
    data-automation="first-name-field"
    id="testCheckbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="Checkbox__label"
    htmlFor="testCheckbox"
  >
    <div
      className="Checkbox__checkbox"
    >
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isHover"
      />
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isSelected"
      />
    </div>
    <Text
      baseline={false}
      className="Checkbox__labelText"
      component="span"
      raw={true}
    >
      Still in role
    </Text>
  </label>
  <FieldMessage
    id="testCheckbox-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`Checkbox should not render the invalid style if tone is set to positive 1`] = `
<div
  className="Checkbox__root"
>
  <input
    checked={false}
    className="Checkbox__input Checkbox__input_isCheckbox"
    id="testCheckbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="Checkbox__label"
    htmlFor="testCheckbox"
  >
    <div
      className="Checkbox__checkbox"
    >
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isHover"
      />
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isSelected"
      />
    </div>
    <Text
      baseline={false}
      className="Checkbox__labelText"
      component="span"
      raw={true}
    >
      Still in role
    </Text>
  </label>
  <FieldMessage
    id="testCheckbox-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
    tone="positive"
  />
</div>
`;

exports[`Checkbox should render the invalid style for tone="critical" even if the field is valid 1`] = `
<div
  className="Checkbox__root Checkbox__invalid"
>
  <input
    checked={false}
    className="Checkbox__input Checkbox__input_isCheckbox"
    id="testCheckbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="Checkbox__label"
    htmlFor="testCheckbox"
  >
    <div
      className="Checkbox__checkbox"
    >
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isHover"
      />
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isSelected"
      />
    </div>
    <Text
      baseline={false}
      className="Checkbox__labelText"
      component="span"
      raw={true}
    >
      Still in role
    </Text>
  </label>
  <FieldMessage
    id="testCheckbox-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
    tone="critical"
  />
</div>
`;

exports[`Checkbox should render the invalid style for valid={false} where no tone is set. It should also pass the valid prop 1`] = `
<div
  className="Checkbox__root Checkbox__invalid"
>
  <input
    checked={false}
    className="Checkbox__input Checkbox__input_isCheckbox"
    id="testCheckbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="Checkbox__label"
    htmlFor="testCheckbox"
  >
    <div
      className="Checkbox__checkbox"
    >
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isHover"
      />
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isSelected"
      />
    </div>
    <Text
      baseline={false}
      className="Checkbox__labelText"
      component="span"
      raw={true}
    >
      Still in role
    </Text>
  </label>
  <FieldMessage
    id="testCheckbox-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
    valid={false}
  />
</div>
`;

exports[`Checkbox should render with button style 1`] = `
<div
  className="Checkbox__root"
>
  <input
    checked={false}
    className="Checkbox__input Checkbox__input_isButton"
    id="testCheckbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="Checkbox__label"
    htmlFor="testCheckbox"
  >
    <Text
      baseline={false}
      className="Checkbox__button"
      component="span"
      raw={true}
    >
      Still in role
    </Text>
  </label>
  <FieldMessage
    id="testCheckbox-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`Checkbox should render with className 1`] = `
<div
  className="Checkbox__root testClassname"
>
  <input
    checked={false}
    className="Checkbox__input Checkbox__input_isCheckbox"
    id="testCheckbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="Checkbox__label"
    htmlFor="testCheckbox"
  >
    <div
      className="Checkbox__checkbox"
    >
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isHover"
      />
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isSelected"
      />
    </div>
    <Text
      baseline={false}
      className="Checkbox__labelText"
      component="span"
      raw={true}
    >
      Still in role
    </Text>
  </label>
  <FieldMessage
    id="testCheckbox-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`Checkbox should render with simple props 1`] = `
<div
  className="Checkbox__root"
>
  <input
    checked={false}
    className="Checkbox__input Checkbox__input_isCheckbox"
    id="testCheckbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="Checkbox__label"
    htmlFor="testCheckbox"
  >
    <div
      className="Checkbox__checkbox"
    >
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isHover"
      />
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isSelected"
      />
    </div>
    <Text
      baseline={false}
      className="Checkbox__labelText"
      component="span"
      raw={true}
    >
      Still in role
    </Text>
  </label>
  <FieldMessage
    id="testCheckbox-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;

exports[`Checkbox should render with standard checkbox style 1`] = `
<div
  className="Checkbox__root"
>
  <input
    checked={false}
    className="Checkbox__input Checkbox__input_isCheckbox"
    id="testCheckbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="Checkbox__label"
    htmlFor="testCheckbox"
  >
    <div
      className="Checkbox__checkbox"
    >
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isHover"
      />
      <CheckMarkIcon
        className="Checkbox__checkMark"
        svgClassName="Checkbox__checkMarkSvg Checkbox__checkMarkSvg_isSelected"
      />
    </div>
    <Text
      baseline={false}
      className="Checkbox__labelText"
      component="span"
      raw={true}
    >
      Still in role
    </Text>
  </label>
  <FieldMessage
    id="testCheckbox-message"
    message=""
    messageProps={
      Object {
        "critical": false,
        "positive": false,
        "secondary": false,
      }
    }
  />
</div>
`;
